src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -302,34 +304,45 @@ private boolean runRepair() {
 
       LazyInitializer<Set<String>> busyHosts = new BusyHostsInitializer(cluster);
 
-      // If we're using a distributed storage, we need to synchronize with other Reaper instances
-      // So we don't start too many segments at the same time
-      if (!lockSegmentRunners()) {
-        LOG.debug(
-            "Not allowed to run the segment for now as another Reaper holds the lock for repair run {}. "
-            + "Will try again later",
-            segment.getRunId());
-        return false;
-      }
       if (!canRepair(segment, keyspace, coordinator, cluster, busyHosts)) {
         LOG.info(
-            "Cannot run segment {} for repair {} at the moment. Will try again later",
-            segmentId,
-            segment.getRunId());
-        SEGMENT_RUNNERS.remove(segment.getId());
+            "Cannot run segment {} for repair {} at the moment. Will try again later", segmentId, segment.getRunId());
+
         try {
           Thread.sleep(SLEEP_TIME_AFTER_POSTPONE_IN_MS);
-        } catch (InterruptedException e) {
-          LOG.debug("Interrupted while sleeping after a segment was postponed... weird stuff...");
-        }
-        releaseSegmentRunners();
+        } catch (InterruptedException ignore) { }
         return false;
       }
 
       try (Timer.Context cxt1 = context.metricRegistry.timer(metricNameForRepairing(segment)).time()) {
         try {
           LOG.debug("Enter synchronized section with segment ID {}", segmentId);
           synchronized (condition) {
+            if (!lockSegmentRunners()) {
```

Comment:
I don't see the corresponding `releaseSegmentRunners()` call in this commit. What did you have in mind about this?

Reply:
the code is inside the `try-finally` block. see https://github.com/thelastpickle/cassandra-reaper/pull/669/commits/5a4ede8c187714c8cfe1dcfb4522fe37d17e5d3e#diff-e6aa6503be815f976c3ed7a845621f4aR387

Reply:
Cool, I missed it.